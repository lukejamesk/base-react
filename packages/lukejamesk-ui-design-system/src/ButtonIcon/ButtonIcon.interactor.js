import { ButtonBaseInteractor } from '@lukejamesk/core'

export default ButtonBaseInteractor
