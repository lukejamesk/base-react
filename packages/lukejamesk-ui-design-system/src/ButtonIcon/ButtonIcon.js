import React from 'react'
import PropTypes from 'prop-types'
import { useStyles } from '@lukejamesk/styles'
import { Button, Icon } from '@lukejamesk/core'

const styles = (theme) => ({
  text: {
    marginLeft: theme.spacing.xs,
  },
})

const variantToIconVariant = {
  default: 'dark',
  primary: 'light',
  secondary: 'primary',
}

const ButtonIcon = ({ icon, variant, children, ...props }) => {
  const classes = useStyles(styles)

  return (
    <Button variant={variant} {...props}>
      <Icon variant={variantToIconVariant[variant]} name={icon} />
      {children && <span className={classes.text}>{children}</span>}
    </Button>
  )
}

ButtonIcon.propTypes = {
  icon: PropTypes.string.isRequired,
  variant: PropTypes.oneOf(['primary', 'secondary', 'default']),
  children: PropTypes.node,
}

export default ButtonIcon
