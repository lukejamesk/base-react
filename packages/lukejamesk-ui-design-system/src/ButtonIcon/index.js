export { default, buttonIconVariants } from './ButtonIcon'
