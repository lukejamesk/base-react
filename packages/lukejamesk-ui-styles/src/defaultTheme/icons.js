const icons = {
  s: 12,
  m: 24,
  l: 48,
}

export default icons
