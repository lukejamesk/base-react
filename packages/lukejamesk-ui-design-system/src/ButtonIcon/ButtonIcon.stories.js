import React from 'react'
import { text, select } from '@storybook/addon-knobs'
import { coffee } from '@lukejamesk/icons'
import { library } from '@lukejamesk/core'
import ButtonIcon from './ButtonIcon'

library.add(coffee)

export default {
  title: 'Design System|Button Icon',
}

const variantsKnob = () => select('Variant', ['default', 'primary', 'secondary'], 'default')

export const Default = () => <ButtonIcon variant={variantsKnob()} icon="coffee" />

export const WithText = () => (
  <ButtonIcon variant={variantsKnob()} icon="coffee">
    {text('Text', 'My button')}
  </ButtonIcon>
)
