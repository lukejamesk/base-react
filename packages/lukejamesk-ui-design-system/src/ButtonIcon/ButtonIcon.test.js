import React from 'react'
import { render } from 'enzyme'
import { library } from '@lukejamesk/core'
import { coffee } from '@lukejamesk/icons'
import ButtonIcon from './ButtonIcon'
import ButtonIconInteractor from './ButtonIcon.interactor'

library.add(coffee)

describe('<ButtonIcon />', () => {
  it('Should render with text', () => {
    const wrapper = render(<ButtonIcon icon="coffee">My content for button</ButtonIcon>)
    expect(wrapper).toMatchSnapshot()
  })
  it('Should render without text', () => {
    const wrapper = render(<ButtonIcon icon="coffee" />)
    expect(wrapper).toMatchSnapshot()
  })

  describe('Actions:', () => {
    let clickSpy
    let interactor
    beforeEach(() => {
      clickSpy = jest.fn()
      interactor = ButtonIconInteractor(
        <ButtonIcon icon="coffee" onClick={clickSpy}>
          My content for button
        </ButtonIcon>,
      )
    })
    it('should fire onClick callback', () => {
      interactor.actions.click()
      expect(clickSpy.mock.calls.length).toEqual(1)
    })
  })
})
