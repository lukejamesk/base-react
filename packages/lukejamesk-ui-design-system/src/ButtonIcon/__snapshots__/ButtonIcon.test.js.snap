// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonIcon /> Should render with text 1`] = `
<button
  class="ljk-bg-gray-200 ljk-hover:bg-gray-400 ljk-p-xs ljk-rounded ljk-uppercase ljk-text-base"
  tabindex="0"
  type="button"
>
  <span>
    <span
      class="ljk-text-black"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-coffee fa-w-20 ljk-fill-current"
        data-icon="coffee"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 640 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M192 384h192c53 0 96-43 96-96h32c70.6 0 128-57.4 128-128S582.6 32 512 32H120c-13.3 0-24 10.7-24 24v232c0 53 43 96 96 96zM512 96c35.3 0 64 28.7 64 64s-28.7 64-64 64h-32V96h32zm47.7 384H48.3c-47.6 0-61-64-36-64h583.3c25 0 11.8 64-35.9 64z"
          fill="currentColor"
        />
      </svg>
    </span>
    <span
      class="ljk-ml-s"
    >
      My content for button
    </span>
  </span>
</button>
`;

exports[`<ButtonIcon /> Should render without text 1`] = `
<button
  class="ljk-bg-gray-200 ljk-hover:bg-gray-400 ljk-p-xs ljk-rounded ljk-uppercase ljk-text-base"
  tabindex="0"
  type="button"
>
  <span>
    <span
      class="ljk-text-black"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-coffee fa-w-20 ljk-fill-current"
        data-icon="coffee"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 640 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M192 384h192c53 0 96-43 96-96h32c70.6 0 128-57.4 128-128S582.6 32 512 32H120c-13.3 0-24 10.7-24 24v232c0 53 43 96 96 96zM512 96c35.3 0 64 28.7 64 64s-28.7 64-64 64h-32V96h32zm47.7 384H48.3c-47.6 0-61-64-36-64h583.3c25 0 11.8 64-35.9 64z"
          fill="currentColor"
        />
      </svg>
    </span>
  </span>
</button>
`;
